### DISCUSSION TITLE
User Interface Recommendations

### DISCUSSION THREAD

**Discussion by nhielost** [nhielost/obs-midi-mg#89] [Ideas]
Please post here about any ideas or recommendations concerning the new user interface for the upcoming 3.0.0 release. Try to answer questions like:

- What do I want to see when I use the plugin?
- Is my idea/recommendation accessible and reasonable?
- Does my idea/recommendation introduce or take away functionality?
- Is my idea/recommendation visually acceptable?

Disclaimer: Not all suggestions will be included. But I would like to know what the users of the plugin want it to look like, so don't be afraid to voice your opinion.

**Comment by azido** [comment]
Hey there,

thanks for this discussion board.

That interface is not as easy to understand as you hoped it will be. On 2.3.0 it was WAY easier to understand: add a binding on the left, name it, then choose the trigger and the wanted actions on the right. That was perfect (at least for me). **The only change this needs is the ability to group bindings in nameable folders** so you don't have to scroll that much if you add a new one - and you could have a better oversight what's configured where.

Now on 3.0.0 things start getting overcomplicated. On 2.3.0 all depends on the given binding, now you divided binding, message and action to different fields. I can't imagine a situation where i would say "oh, ok, lets switch that action to another binding" or "i wish, i could give that assignment a name". But ok, i think after a while we can handle that.

But now for my recommendations/ideas:

1. As before, i wish we could combine/store bindings in separate folders the way OBS lets you group sources together in one folder.  I for myself use 2 controllers - a novation remote zero sl and an akai mpd18 - with many knobs and buttons to assign, so naturally i have many bindings. 
2.  In the bindings window: you served edit options on the right. That's cool, but the only option to edit is to SELECT one of the given devices / messages / actions. I wish i could do more once i click on edit: rename the message/action, double click on it's name to go directly to it's corresponding entry in the other tabs. Example: untitled message (3) - F2 to rename, double click jumps to the corresponding config in the messages tab. This comes handy especially when we have a lot of imported bindings, as they are all named "untitled" also in messages/actions. Right now i have to remember the name, then switch to the messages tab, then search for that name and select it again to see it's entries.
3.  Cosmetic change: that icon for the devices.. maybe change it to a keyboard/piano icon instead of a monitor? 

Beside that i have some technical issues, i'll adress them on another post.

Finally: thank you for your great plugin, i really appreciate your work and your try to improve it further!

**Reply by azido** [reply]
Me again, while trying to reproduce my issue i found some more enhancement wishes.
When adding a new binding, i naturally use the + sign on the left. I give it a name, then use the edit button to select the device. The logical next step is to add a message, so i click on the edit sign beside the message - but from there, i can only select already available messages. The + sign still exists, but does not work on that page, i cannot add a new message here. This really interrupts the flow as now i first have to switch to the messages tab to add one here. Please make that usable here. We don't need to dive deeper into the configuration here, but it's more conveniant when we can add an empty new message/action from the selection here and give it a suitable name which we then can bring to life in the other tabs. Hope i described good enough what i mean.

**Comment by nhielost** [comment]
This discussion is being closed due to the implementation of the issue in question in version 3.0.0.

Thanks for your help!
